corgidrp/calibrate_kgain.py

```diff
@@ -0,0 +1,1099 @@
+    also creates a mean pupil array from a separate stack of frames of uniform 
+    exposure time. The mean pupil array is scaled to the mean of each stack and 
+    statistics (mean and std dev) are calculated for bins from the frames in it. 
+    kgain (e-/DN) is calculated from the means and variances within the 
+    defined minimum and maximum mean values. A photon transfer curve is plotted 
+    from the std dev and mean values from the bins. 
+    
+    Parameters
+    ----------
+    stack_arr : array-like
+        The stack of stacks of EXCAM illuminated pupil L1 SCI frames (counts 
+        in DN) having a range of exp times. stack_arr contains a stack of 
+        stacks, and all sub-stacks must have the same number of frames, which 
+        is a minimum of 5. The frames in a sub-stack must all have the same 
+        exposure time time. Must have at least 10 sub-stacks. (More than 20 
+        sub-stacks recommended. The mean signal in the pupil region should 
+        span from about 100 to about 10000 DN. note: unity em gain is 
+        recommended when k-gain is the primary desired product, since it is 
+        known more accurately than non-unity values.)
+    
+    stack_arr2 : array-like
+        The stack of EXCAM illuminated pupil L1 SCI frames (counts in DN). 
+        stack_arr2 contains a stack of at least 30 frames of uniform exposure 
+        time, such that the net mean counts in the pupil region is a few thousand 
+        DN (2000 to 4000 DN recommended; note: unity em gain is recommended 
+        when k-gain is the primary desired product, since it is known more 
+        accurately than non-unity values. stack_arr and stack_arr2 must be 
+        obtained under the same positioning of the pupil relative to the 
+        detector.
+    
+    emgain : float
+        The value of the measured/actual EM gain used to collect the frames used 
+        to build the stack_arr and stack_arr2 arrays. Must be >= 1.0. (note: 
+        unity em gain is recommended when k-gain is the primary desired product, 
+        since it is known more accurately than non-unity values.)
+    
+    min_val : int
+        Minimum value (in DN) of mean values from sub-stacks to use in calculating 
+        kgain. (> 400 recommended)
+        
+    max_val : int
+        Maximum value (in DN) of mean values from sub-stacks to use in calculating 
+        kgain. Choose value that avoids large deviations from linearity at high 
+        counts. (< 6,000 recommended)
+    
+    binwidth : int
+        Width of each bin for calculating std devs and means from each 
+        sub-stack in stack_arr. Maximum value of binwidth is 800. NOTE: small 
+        values increase computation time.
+        (minimum 10; binwidth between 65 and 75 recommended)
+    
+    mkplot : boolean
+        Option to display plots. Default is None. If mkplot is anything other 
+        than None, then this option is chosen.
```

Comment:
@maxwellmb if the output directory can be freely defined and someone defines it within corgidrp repo, it will be tracked by git and produce annoying 'files changed'. Is this a possibility that you have considered?

Reply:
I hadn't considered that, so feel free to direct it somewhere else. This will normally be set by the walker when we pick an output directory for files.

Reply:
Done